internal/setup/setup.go

```diff
@@ -410,27 +430,52 @@ func (yp *yamlPath) SameContent(other *yamlPath) bool {
 }
 
 type yamlArch struct {
-	list []string
+	List []string
 }
 
 func (ya *yamlArch) UnmarshalYAML(value *yaml.Node) error {
 	var s string
 	var l []string
 	if value.Decode(&s) == nil {
-		ya.list = []string{s}
+		ya.List = []string{s}
 	} else if value.Decode(&l) == nil {
-		ya.list = l
+		ya.List = l
 	} else {
 		return fmt.Errorf("cannot decode arch")
 	}
 	// Validate arch correctness later for a better error message.
 	return nil
 }
 
+func (ya yamlArch) MarshalYAML() (interface{}, error) {
```

Comment:
Why is this method on the value, while yamlPath is on a pointer?  How is the decision being made?

Reply:
Agree, let's use a pointer receiver here.

Reply:
We need this method on the value because it's a value field inside `yamlPath`. Otherwise, the `MarshalYAML` method on pointer will not be called.

```go
type yamlPath struct {
	Arch     yamlArch     `yaml:"arch,omitempty"`
	...
}
```